csharp/examples/IoTExample/Model/SampleDataset.cs

```diff
@@ -0,0 +1,211 @@
+        private readonly MemoryAllocator _memoryAllocator;
+
+        public Dictionary<string, string> activityLabel = new Dictionary<string, string>()
+            {
+                {"walking", "A"},
+                {"jogging", "B"},
+                {"stairs", "C"},
+                {"sitting", "D"},
+                {"standing", "E"},
+                {"typing", "F"},
+                {"teeth", "G"},
+                {"soup", "H"},
+                {"chips", "I"},
+                {"pasta", "J"},
+                {"drinking", "K"},
+                {"sandwich", "L"},
+                {"kicking", "M"},
+                {"catch", "O"},
+                {"dribbling", "P"},
+                {"writing", "Q"},
+                {"clapping", "R"},
+                {"folding", "S"},
+            };
+
+        public SampleDataset(int inputs, int capacity)
+        {
+            _inputs = inputs;
+            _capacity = capacity;
+            _rows = new BlockingCollection<SensorData>(_capacity);
+
+            _colSubjectId = new List<int>(capacity);
+            _colActivityLabel = new List<string>();
+            _colTimestamp = new List<long>();
+            _colXAxis = new List<double>();
+            _colYAxis = new List<double>();
+            _colZAxis = new List<double>();
+
+            _recordBatches = new List<RecordBatch>();
+            _memoryAllocator = new NativeMemoryAllocator(alignment: 64);
+        }
+
+        public void Produce()
+        {
+            Random rand = new Random();
+            bool success;
+
+            List<string> keyList = new List<string>(activityLabel.Keys);
+            int count = keyList.Count;
+
+            DateTime now = DateTime.Now;
+            long unixTime = ((DateTimeOffset)now).ToUnixTimeSeconds();
+
+            while (_size < _inputs)
+            {
+                string randomKey = keyList[rand.Next(count)];
+                string label = activityLabel[randomKey];
+
+                // generate missing values
+                if (rand.Next(10_000) == 9_999)
+                {
+                    label = null;
+                }
+
+                success = _rows.TryAdd(new SensorData
+                {
+                    subjectId = rand.Next(1000, 2001),
+                    activityLabel = label,
+                    timestamp = unixTime++,
+                    x_Axis = rand.NextDouble(),
+                    y_Axis = rand.NextDouble(),
+                    z_Axis = rand.NextDouble(),
+                }); 
+
+                if (success)
+                {
+                    //Console.WriteLine($"Enqueue Task 0");
+                    _size++;
+                }
+                else
+                {
+                    Console.WriteLine("Producing is blocked, percent completed is: {0}%", Math.Round((double)_size / _inputs, 4) * 100);
+                }
+            }
+
+            _rows.CompleteAdding();
+        }
+
+        public async void Consume()
```

Comment:
This method doesn't need to be `async` since it doesn't `await` anything.

Also - avoid `async void` methods in general. Always return `async Task`, if you can.

Reply:
Done.